### DISCUSSION TITLE
endpoint do pobrania listy 'Moje zakupy'

### DISCUSSION THREAD

**Discussion by KubaMiszcz** [allegro/allegro-api#5394] [Suggestion - verification]
udostepnienie endpointu do pobrania listy 'Moje zakupy', szczegoly zakupu/zwrotu itp - z punktu widzenia kupujacego

**Comment by Przemko92** [comment]
Byłoby to przydatne np. do integracji z systemami smart home, jak HomeAssistant. Chętnie dodałbym obsługę paczek z Allegro do mojego smart domu(informcje o statusie, prawdopodobnej dacie dostarczenia itp.). Jest taki endpoint, który zwraca wszystkie potrzebne dane - https://api.allegro.pl/myorder-api/myorders/ niestety nie można się do niego dostać używając auth token, tylko z użyciem cookies zalogowanych w przeglądarce.

**Comment by gkeb** [comment]
Opcja informacjo o wysyłce już kupionych produktów trafiających do HA to ciekawa sprawa. Szczególnie przy tych produktach, które nie są wysyłane od razu. Inna sprawa to podgląd do tego może mieć cała rodzina, a nie tylko właściciel konta na allegro. Ktoś będący w domu może się spodziewać, że przyjdzie przesyłka, nawet jak zamawiający zapomniał powiedzieć. Bezpieczniej.

**Comment by anglerCezary** [comment]
Tego brakuje. Chętnie widział bym taką funkcjonalność API.

**Comment by artpcpl** [comment]
Genialny pomysł,  podnoszę obydwie ręce w górę ✪✪✪✪✪

**Comment by rafiej1** [comment]
Super pomysł . Tego brakuje :)

**Comment by cichy2k** [comment]
Przyłaczam sie do prośby gdyż jestem stałem użytkownikiem tego serwisu i taka opcja bardzo wzbogaciła by muj system zarządzania domem o super opcję

**Comment by hahyes** [comment]
> kupujacego

Problem niestety jest taki, że API jest przeznaczone głównie dla sprzedawców i takie prośby niestety mają od zerowego po niski priorytet. O ile w ogóle jest w ogóle to rozpatrywane. Ale może jakiś pracownik w wolnym czasie znajdzie na to moment... któż to wie.

**Reply by Przemko92** [reply]
Tu bardziej chodzi o zainteresowanie. Mało który użytkownik (poza sprzedawcami) dotychczas potrzebował takiego api. Jeśli będzie zainteresowanie, Allegro doda funckję. Jak widać zainteresowanie jest. :)

**Comment by KubaMiszcz** [comment]
tak jak piszesz, nie generuje to wprost zysku, a nie wchodzisz na strone wiec nie ogladasz reklam/propozycji. Kiedys to podobno bylo, ale moze wypadlo po jakiejs reorganizacji a dzis ma lowest priority;]
moze jakis stazysta to zrobi kiedys albo praktykant

**Reply by Przemko92** [reply]
Kiedyś było w starym web api, potem podobno wyleciało z powodu małego zainteresowania. Świat się zmienia, teraz zainteresowanie byłoby większe.

**Comment by Jaryn91** [comment]
Tworzę system do zarządzania domowym budżetem. Wszystkie zakupy z allegro w wyciągach bankowych są zapisane jako 'Allegro'. Użytkownik musi iść do strony allegro i sprawdzać organoleptycznie jakie zakupy robił w konkretnych dniach i dopasować kategorię ręcznie - co kupił do odpowiedniej kategorii dla konkretnej transakcji. Owszem można napisać program, który będzie wchodzić na stronę i parsować treści... Ale pewniejszy będzie endpoint. Mają endpointa można byłoby zautomatyzować pobieranie danych o zakupach i mógłbym zautomatyzować dodawanie kategorii do konkretnych zakupów.

**Comment by niqu1982** [comment]
śledzenie i powiadomienie w api dla odbiorcy - bardzo by się przydało !

**Comment by danielskowronski** [comment]
Dostęp do tego API to nie tylko kwestia domowych "power userów" ustawiających sobie magiczne automatyzacje, ale także twórców aplikacji do śledzenia przesyłek od różnych firm kurierskich, którzy chcieliby zintegrować Allegro - chociażby [Parcel](https://parcelapp.net) czy [ParcelsApp](https://parcelsapp.com) mają Amazona, AliExpress i kilka innych. 

Na ten moment aplikacja Allegro na telefony nie nadaje się nawet do śledzenia rzeczy kupionych na samym Allegro - ot, kolejne miejsce spamujące powiadomieniem "odbierz paczkę", a nie pozwalającym na podgląd wszystkich paczek (żeby ocenić czy jakaś nie utknęła) czy wygodne odbieranie z automatu paczkowego dowolnej marki. Poza tym nawet użytkownicy kupujący absurdalne ilości produktów na platformie czasem zamawiają rzeczy gdzie indziej i chcieliby mieć możliwość sprawdzenia przesyłek w *jednym* miejscu. 

Dlatego jest bardzo smutne, że endpointy `https://api.allegro.pl/myorder-api/myorders` albo nie używają OAuth, albo Allegro agresywnie ukrywa scope'y niezbędne do ich użycia.

**Comment by dawidof** [comment]
Przydało się, przynajmniej trzymać historię lokalnie dłużej, bo po 2 latach allegro czyści historię zamówień, a opisów produktów czasami jeszcze szybciej. Smutne to jest

**Comment by luqasz** [comment]
Przydałby się taki dostęp dla zwykłego kupującego.

**Comment by pciesielski32** [comment]
Dorzucam przypadek, który nie padł jeszcze w tym wątku, a pokazuje lukę nie do obejścia z zewnątrz.

Powiadomienia mailowe nie zastępują dostępu do własnych zamówień, bo nie każdy przewoźnik cokolwiek wysyła. Przy dostawie do automatów ORLEN Paczka na moją skrzynkę nie przychodzi ani wiadomość o gotowości do odbioru, ani o odbiorze. Powiadomienia z Allegro kończą się na informacji, że przesyłka ruszyła w drogę. Efekt jest taki, że po stronie kupującego nie da się ustalić, czy paczka faktycznie została odebrana, dopóki ktoś nie zajrzy w konto ręcznie. Przy zakupach rozliczanych na firmę i przy prowadzeniu własnego rejestru sprzętu ma to znaczenie, bo data wydania rzeczy wyznacza bieg rękojmi.

Druga sprawa dotyczy już samego Allegro. Brak takiego zasobu nie sprawia, że ludzie przestają pobierać własne dane, tylko że robią to przeglądarką sterowaną automatem. To jest zawodne dla użytkownika i obciążające dla waszych zabezpieczeń, bo ruch, który mógłby iść przez uwierzytelniony i limitowany interfejs, ląduje na zwykłych stronach i wygląda jak nadużycie. Nawet bardzo wąski zakres, czyli lista zamówień z datą, statusem dostawy i numerem przesyłki, bez danych sprzedających i bez cen, rozwiązałby większość opisanych w tym wątku potrzeb.

Wątek ma cztery i pół roku oraz siedemdziesiąt kilka głosów. Czy moglibyśmy dostać informację, na czym stoi weryfikacja i czego brakuje, żeby przesunąć go dalej albo domknąć?

**Reply by PrzemyslawLukanowski** [reply]
Skontaktowałem się ponownie z zespołem odpowiedzialnym za zakładkę Moje zakupy - wrócę z informacją, jak sytuacja wygląda obecnie.

**Reply by PrzemyslawLukanowski** [reply]
Na tę chwilę pozostawiamy sugestię w statusie "Verification" - do końca roku nie uda nam się zaimplementować endpointu, nie wykluczamy jednak, że będzie to możliwe w pierwszych kwartałach następnego. Wrócimy z informacją, gdy będziemy mogli podzielić się większą ilością szczegółów.
